### DISCUSSION TITLE
Why buffer sizes are 65535 in your I/O core?

### DISCUSSION THREAD

**Discussion by ghost** [mitmproxy/mitmproxy#5931] [Q&A]
Is there a reason why you use buffer sizes 65535 apparently all across your core proxy I/O? I am interested in hearing why or otherwise how to choose a good buffer size

![image](https://user-images.githubusercontent.com/118858207/219029225-a3b5ced4-8089-40f9-a35a-010cf1fb6bcb.png)

https://github.com/mitmproxy/mitmproxy/blob/main/mitmproxy/proxy/server.py#L284

**Comment by mhils** [comment]
We haven't evaluated this in any rigorous way, but IIRC I picked 65535 because that's approximately a power of 2 and the maximum size of a TCP packet. The Python docs recommend small powers of two, we're a bit on the bigger side here because we do a lot of processing per data chunk relatively speaking. I'd be curious to hear if picking something smaller/bigger makes any positive or negative impact!
